eserde/tests/default.rs

```diff
@@ -0,0 +1,22 @@
+#[derive(eserde::Deserialize)]
+struct Foo {
+    #[serde(rename = "route", default = "default_route")]
+    route_0: String,
+    #[serde(default = "default_route")]
+    route_1: String,
+}
+
+fn default_route() -> String {
+    "/".to_string()
+}
+
+#[test]
+fn test() {
+    let x: Foo = eserde::json::from_str(r#"{}"#).unwrap();
+    assert_eq!("/", x.route_0);
+    assert_eq!("/", x.route_1);
+
+    let x: Foo = eserde::json::from_str(r#"{"route": "/dev/null"}"#).unwrap();
+    assert_eq!("/dev/null", x.route_0);
+    assert_eq!("/", x.route_1);
+}
```

Comment:
This doesn't really exercise the logic we care about, since deserialization succeeds.
We should try to deserialize for errors and then verify that the error buffer is empty.

Reply:
We need to test the happy case, but we also need the unhappy case.

Reply:
Good point, have updated the tests to include the actual error path now